### PR TITLE
Add NSCountedSet Category

### PR DESCRIPTION
**Pull request by theocalmes**
and added docs to a few classes

### CODE REVIEW

File: BotKit/Categories/NSCountedSet+ObjectAccess.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ @return Set object.
+ */
+- (id)objectWithLowestCount;
+
+/** Returns an array with the top N most counted objects from the set. Ordered by count value.
+ 
+ @param numberOfObjects Number of objects returned.
+ @return Array of objects.
+ */
+- (NSArray *)objectsWithHighestCounts:(NSInteger)numberOfObjects;
```
Comment (hyperspacemark) [inline BotKit/Categories/NSCountedSet+ObjectAccess.h:51]: How about `-arrayOfMostCommonObjectsWithLimit:`?

File: BotKit/Categories/NSCountedSet+ObjectAccess.h [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+ @return All counts for all objects.
+ */
+- (NSInteger)countAllObjects;
+
+/** Returns an array of all the ojects. An object with a count of 10 will have 10 copies in the returned array.
+
+ @return Array of size countAllObjects containing objects.
+ */
+- (NSArray *)allObjectsFlattened;
```
Comment (hyperspacemark) [inline BotKit/Categories/NSCountedSet+ObjectAccess.h:67]: I think `-flattenedObjects` would suffice.

File: BotKit/Categories/NSCountedSet+ObjectAccess.m [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+//
+
+#import "NSCountedSet+ObjectAccess.h"
+#import "NSArray+ObjectAccess.h"
+
+@implementation NSCountedSet (ObjectAccess)
+
+#pragma mark - Function Helpers
+
+NSInteger comp(const void *i, const void *j)
```
Comment (hyperspacemark) [inline BotKit/Categories/NSCountedSet+ObjectAccess.m:16]: C functions should be named like classes are; with a prefix and camel-case. Also, please expand `comp` to `compare`.

File: BotKit/Categories/NSCountedSet+ObjectAccess.m [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+{
+    return *(NSInteger *)i - *(NSInteger *)j;
+}
+
+#pragma mark - Public Methods
+
+- (id)randomObject
+{
+    NSArray *setArr = [self allObjects];
```
Comment (hyperspacemark) [inline BotKit/Categories/NSCountedSet+ObjectAccess.m:25]: Minor nit but could this be `setArray`?

**Comment by hyperspacemark** [comment]
Couple comments, @theocalmes, but otherwise looks good. :+1:

**Comment by Wisors** [comment]
There is a bug in `arrayOfMostCommonObjectsWithLimit`. In case of some words having the same count your method return array with duplicates of one of this words. ('indexOfObject:@(num)' return the same word).

```
 NSInteger num = countsCArray[i];
 NSInteger index = (NSInteger)[counts indexOfObject:@(num)];
 [result addObject:objs[index]]
```

For example, I have a large string and I expect the next result:
...
      {
        count = 18;
        word = a;
      },{
        count = 18;
        word = that;
     }, {
        count = 18;
        word = was;
     }
...
but instead this method return:
...
was,
was,
was,
...

**Pull request closed without merging**